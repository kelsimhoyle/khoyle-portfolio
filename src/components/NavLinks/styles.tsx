import styled from "styled-components";


export const NavUl = styled.ul`
    display: flex;
    flex-direction: row;
    
    li {
        list-style: none;
        padding: 1rem;
    }
`