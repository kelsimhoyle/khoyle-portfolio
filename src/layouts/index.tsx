import * as React from "react";
import { useStaticQuery, graphql } from "gatsby";
import "@fontsource/montserrat";
import "@fontsource/biorhyme";
import "../styles/reset.css";

import { GlobalStyle } from "../styles/utils";

import Header from "../components/Header";
import Footer from "../components/Footer";
import { siteMetadata } from "../../gatsby-config";
import { StringDecoder } from "string_decoder";
// import "./layout.css"

interface LayoutPropsInterface {
  children: React.ReactNode
}

export type MenuLinksType = {
    name: string;
    link: string;
}

type DataType = {
    site: {
      siteMetadata: {
        title: string;
        menuLinks: MenuLinksType[];
      }
    }
}

const Layout = ({ children }: LayoutPropsInterface) => {
  const data: DataType = useStaticQuery(graphql`
  query SiteTitleQuery {
    site {
      siteMetadata {
        title
        menuLinks {
            name
            link
       }
      }
    }
  }
  `)

  const siteTitle: string = data.site.siteMetadata?.title || `Title`;
  const menuLinks: MenuLinksType[] = data.site.siteMetadata.menuLinks;

  return (
    <>
      <GlobalStyle />
        <Header
          siteTitle={siteTitle}
          menuLinks={menuLinks}
        />      
        <main>{children}</main>
        <Footer
          siteTitle={data.site.siteMetadata.title}
          menuLinks={data.site.siteMetadata.menuLinks}
        />
    </>
  )
}

export default Layout
