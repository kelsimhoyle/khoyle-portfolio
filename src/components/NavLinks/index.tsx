import React from "react";
import { Link } from "gatsby";
import { PrimaryLink } from "../../styles";
import { MenuLinksType } from "../../layouts";
import { NavUl } from "./styles";

type NavLinksProps = {
    menuLinks: MenuLinksType[]
}

const NavLinks = ({ menuLinks }: NavLinksProps) => {
    return (
        <NavUl>
            {menuLinks.map(link => (
                <li key={link.name}>
                    <PrimaryLink>
                        <Link to={link.link}>
                            {link.name}
                        </Link>
                    </PrimaryLink>
                </li>
            ))}
        </NavUl>
    )
}

export default NavLinks;