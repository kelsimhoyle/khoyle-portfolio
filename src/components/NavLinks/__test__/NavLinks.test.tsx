import React, { Children } from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import NavLinks from "..";
import { MenuLinksType } from "../../../layouts"

let getByTestId;
let container;

const data: MenuLinksType[] =  [
    {
      name: "Home",
      link: "/"
    },
    {
      name: "About",
      link: "/about"
    },
    {
      name: "Portfolio",
      link: "/portfolio"
    },
    {
      name: "Contact",
      link: "/contact"
    }
  ];

beforeEach(() => {
    const component = render(<NavLinks menuLinks={data} />);
    getByTestId = component.getByTestId;
    container = component.container;
});

test("contact buttons renders correctly", () => {
    const tree = renderer.create(<NavLinks menuLinks={data} />).toJSON()
    expect(tree).toMatchSnapshot()
});

test("the nav ul has the correct li items", () => {
    const navEl = getByTestId("nav");

    expect(navEl).toHaveProperty("children");
    expect(container.firstChild.childNodes.length).toEqual(data.length);


});

test("each link has the correct href", () => {
    for (let i = 0; i < data.length; i++ ) {
        expect(container.getElementsByClassName("link").item(i)).toHaveAttribute("href", data[i].link)
    }
})