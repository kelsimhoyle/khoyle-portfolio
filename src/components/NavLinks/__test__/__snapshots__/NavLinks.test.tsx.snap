// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact buttons renders correctly 1`] = `
<ul
  className="sc-dmlqKv bezaOw"
  data-testid="nav"
>
  <li>
    <span
      className="sc-iqHYmW frONGD"
    >
      <a
        className="link"
        href="/"
      >
        Home
      </a>
    </span>
  </li>
  <li>
    <span
      className="sc-iqHYmW frONGD"
    >
      <a
        className="link"
        href="/about"
      >
        About
      </a>
    </span>
  </li>
  <li>
    <span
      className="sc-iqHYmW frONGD"
    >
      <a
        className="link"
        href="/portfolio"
      >
        Portfolio
      </a>
    </span>
  </li>
  <li>
    <span
      className="sc-iqHYmW frONGD"
    >
      <a
        className="link"
        href="/contact"
      >
        Contact
      </a>
    </span>
  </li>
</ul>
`;
